src/Microsoft.AspNetCore.ResponseCaching/Internal/LoggerExtensions.cs

```diff
@@ -0,0 +1,300 @@
+    /// </summary>
+    internal static class LoggerExtensions
+    {
+        private static Action<ILogger, string, Exception> _logRequestMethodNotCacheable;
+        private static Action<ILogger, Exception> _logRequestWithAuthorizationNotCacheable;
+        private static Action<ILogger, Exception> _logRequestWithNoCacheNotCacheable;
+        private static Action<ILogger, Exception> _logRequestWithPragmaNoCacheNotCacheable;
+        private static Action<ILogger, TimeSpan, Exception> _logExpirationMinFreshAdded;
+        private static Action<ILogger, TimeSpan, TimeSpan, Exception> _logExpirationSharedMaxAgeExceeded;
+        private static Action<ILogger, TimeSpan, TimeSpan, Exception> _logExpirationMustRevalidate;
+        private static Action<ILogger, TimeSpan, TimeSpan, TimeSpan, Exception> _logExpirationMaxStaleSatisfied;
+        private static Action<ILogger, TimeSpan, TimeSpan, Exception> _logExpirationMaxAgeExceeded;
+        private static Action<ILogger, DateTimeOffset, DateTimeOffset, Exception> _logExpirationExpiresExceeded;
+        private static Action<ILogger, Exception> _logResponseWithoutPublicNotCacheable;
+        private static Action<ILogger, Exception> _logResponseWithNoStoreNotCacheable;
+        private static Action<ILogger, Exception> _logResponseWithNoCacheNotCacheable;
+        private static Action<ILogger, Exception> _logResponseWithSetCookieNotCacheable;
+        private static Action<ILogger, Exception> _logResponseWithVaryStarNotCacheable;
+        private static Action<ILogger, Exception> _logResponseWithPrivateNotCacheable;
+        private static Action<ILogger, int, Exception> _logResponseWithUnsuccessfulStatusCodeNotCacheable;
+        private static Action<ILogger, Exception> _logNotModifiedIfNoneMatchStar;
+        private static Action<ILogger, EntityTagHeaderValue, Exception> _logNotModifiedIfNoneMatchMatched;
+        private static Action<ILogger, DateTimeOffset, DateTimeOffset, Exception> _logNotModifiedIfUnmodifiedSinceSatisfied;
+        private static Action<ILogger, Exception> _logNotModifiedServed;
+        private static Action<ILogger, Exception> _logCachedResponseServed;
+        private static Action<ILogger, Exception> _logGatewayTimeoutServed;
+        private static Action<ILogger, Exception> _logNoResponseServed;
+        private static Action<ILogger, string, string, Exception> _logVaryByRulesStored;
+        private static Action<ILogger, Exception> _logResponseStored;
+        private static Action<ILogger, Exception> _logResponseNotStored;
+        private static Action<ILogger, Exception> _logResponseContentLengthMismatchNotStored;
+
+        static LoggerExtensions()
+        {
+            _logRequestMethodNotCacheable = LoggerMessage.Define<string>(
+                logLevel: LogLevel.Debug,
+                eventId: 1,
+                formatString: "Request is not cacheable due to the HTTP method: {Method}.");
```

Comment:
'Request is not cacheable'... is the request ever cacheable? 'The request cannot be served from cache...'